docs/guides/dataloaders.md

```diff
@@ -171,6 +171,6 @@ class MyGraphQL(GraphQL):
 @strawberry.type
 class Query:
     @strawberry.field
-    async def get_user(self, info: Info, id: strawberry.ID) -> User:
-        return await info.context["user_loader"].load(id)
+    async def get_user(self, id: strawberry.ID) -> User:
+        return await strawberry.context["user_loader"].load(id)
```

Comment:
do you think we can:

1. type it somehow?
2. use context.user_loader instead of context["user_loader"] 

?

Reply:
> 1. type it somehow?

I'm not sure how we could type it unfortunately. You can always cast it when you import it but that's not great. I think that's one of the downsides of this proposal. 

> 2. use context.user_loader instead of context["user_loader"]

That should already work. See tests.

Reply:
yeah, I wonder if doing:

```
ctx: Context[Something] = strawberry.context
```

is annoying or not.

Do we get benefits by using ContextVars and still using parameters (and dependency injection)?

Reply:
> Do we get benefits by using ContextVars and still using parameters (and dependency injection)?

You start defining (and exporting from Strawberry) stand alone functions that get useful things from the context. E.g. `current_user()` The only issue with that is there wouldn't be a way for those functions to be known about early in the execution to register things into the context before resolution happens. E.g. you couldn't do `get_dataloader(UserDataloader)` inside a resolver without also registering an extension on the schema that would inject the dataloader into the context. Dependency injection would allow you to that.

Reply:
Oh it would make the data validation proposal (https://github.com/strawberry-graphql/strawberry/discussions/820#discussioncomment-1081049) a bit cleaner. Instead of having to pass info to the `.validate()` function you could grab the context from the context variable.